sapi/cli/php_cli_server.c

```diff
@@ -1385,11 +1385,12 @@ static void php_cli_server_request_translate_vpath(php_cli_server_request *reque
 	if (request->vpath_len > 0 && request->vpath[0] != '/') {
 		*p++ = DEFAULT_SLASH;
 	}
-	q = request->vpath + request->vpath_len;
-	while (q > request->vpath) {
-		if (*q-- == '.') {
-			is_static_file = 1;
-			break;
+	q = request->vpath;
+	{
+		if (q) {
+			char *dot = strrchr(q, '.');
+			if (dot && !strcmp(dot, ".php"))
```

Comment:
`strlen(dot)` could be less than 4.

Reply:
In such case the file has not a ".php" extension. What's the problem?

Reply:
If both are `\0` terminated, i meant. Should be, actually, so perhaps no worries.

Thanks.